client/config.c

```diff
@@ -192,6 +245,17 @@ TDNFReadConfig(
         }
     }
 
+    pszTdnfVersion = TDNFGetVersion();
+
+    if (pConf->pszOSName == NULL)
+        TDNFAllocateString("UNKNOWN", &pConf->pszOSName);
+
+    if (pConf->pszOSVersion == NULL)
+        TDNFAllocateString("UNKNOWN", &pConf->pszOSVersion);
+
+    dwError = TDNFAllocateStringPrintf(&pConf->pszUserAgentHeader, "User-Agent:tdnf/%s %s/%s", pszTdnfVersion, pConf->pszOSName, pConf->pszOSVersion);
```

Comment:
You need a space after "User-Agent:". But, I think it doesn't need to be part of the string at all. See example at https://curl.se/libcurl/c/CURLOPT_USERAGENT.html 
```
[curl_easy_setopt](https://curl.se/libcurl/c/curl_easy_setopt.html)(curl, [CURLOPT_USERAGENT](https://curl.se/libcurl/c/CURLOPT_USERAGENT.html), "Dark Secret Ninja/1.0");
```

Also make sure it's compliant with RFC 7231, see https://www.rfc-editor.org/rfc/rfc7231#page-46

Also, I don't see anywhere where you are setting the header for curl. At some point you will have to call `curl_easy_setopt(curl, CURLOPT_USERAGENT, ...`